lib/matplotlib/collections.py

```diff
@@ -2111,3 +2111,39 @@ def get_cursor_data(self, event):
         if contained and self.get_array() is not None:
             return self.get_array().ravel()[info["ind"]]
         return None
+
+
+class PolyQuadMesh(PolyCollection, QuadMesh):
```

Comment:
Would it make more sense to reverse the inheritance order, and use "super" below?  Is the PolyQuadMesh more like a QuadMesh than a PolyCollection, falling back on the latter only for its draw method, as is already done explicitly below?  Or is the current order correct with respect to all inherited methods?

Reply:
I think we actually want most of the methods from PolyCollection because we are drawing Polygons and want that `set_paths()` method. We really want the QuadMesh coordinates stored internally is all I think. I reworked the inheritance a bit in the most recent commit.

One thing I need to think about a bit more is how to handle the masking of paths here... `len(self._paths()) != coords[:2].size` when there are masked elements. So, we may actually have to overwrite the `set_paths()` here to make sure that we are updating the proper facecolors to correspond to the mesh.

Currently, we return a `PolyCollection` that has a smaller number of paths, so if you want to update the array with the proper colors you need to account for that resizing yourself. (We do that in Cartopy, so perhaps just porting that upstream here is the way to go)

Reply:
It turns out my understanding of multiple inheritance was, and probably still is, incomplete.  There are arguments in favor of "composition instead of inheritance", and I wonder whether this is an example where using composition--making an internal QuadMesh instance, and referring to its contents explicitly when needed--might be more readable, robust, and maintainable.  It would reduce the inheritance to single--much easier to fit in one's brain.

Reply:
Or, is there a part of QuadMesh code that should be factored out into a mixin?

Reply:
I added a `_MeshData` mixin as a separate commit. Let me know your thoughts on how that organization seems now.

Reply:
The mixin looks fine to me.